Spec.md

```diff
@@ -174,6 +174,19 @@ spec:
         version: provider-key-version
         property: provider-key-property
 
+        # (optional) provider-specific instructions on how to fetch the secret
+        azurekv:
+          # fetch tags and use ref.property to extract a certain tag
+          # tags are included in a GetSecret() respose
+          # see https://github.com/Azure/azure-sdk-for-go/blob/v51.1.0/services/keyvault/v7.0/keyvault/models.go#L2865
+          fetchTags: true
```

Comment:
It seems like fetching metadata from the ExternalSecret and decoding are two common requirements that the ExternalSecret may need functionality to handle. These seem generic enough that they may not need to be nested under their specific provider.

Per [Kubernetes API conventions](https://github.com/kubernetes/community/blob/6254e11696ac0886d554021a27c8179b459502c6/contributors/devel/sig-architecture/api-conventions.md#primitive-types) we should avoid boolean fields and prefer Enum options.

```yaml
  # Data defines the connection between the Kubernetes Secret keys and the Provider data
  data:
    - secretKey: secret-key-to-be-managed
      remoteRef:
        key: provider-key
        version: provider-key-version
        property: provider-key-property

        # Policy for fetching tags/labels from provider secrets, possible options are Fetch, None
        metadataPolicy: Fetch
        # Strategy for decoding provider secrets, options are Binary, Base64, BestEffort (not sure on this or DefaultProvider type)
        decodeStrategy: Binary 
```

Reply:
I tried to outline the concept in pseudocode. Do we align on that? 

```js
data, metadata = fetchSecret(...);

// operate on metadata instead of the actual data
// to extract k/v pairs from there
if ref.metadataPolicy == Fetch:
  data = metadata
  metadata = nil // fallback for strat=BestEffort

if ref.decodeStrategy == Binary:
  // there's no point to extract a ref.Property from binary
  return decodeBin(data)
else if ref.decodeStrategy == Base64:
  data = base64dec(data); // base64 could potentially have a json payload
else if ref.decodeStrategy == BestEffort
  // this can be everything
  // this is where we implement provider-specific decoding strats like PKCS#7/11/12 or whatever
  // a provider can implement custom schemas so users can access their keys
  // from special content-types (for the azure example: PEM encoded key/cert)
  strat = whatStratToUse(data, metadata)
  data = strategies[strat].call(data)

// proceed with extracting information from secret
// i.e. try to parse json, extract key etc.
```

One thing tho: I (as a user) may want to have the raw values from the upstream API. This is currently not supported. We should consider adding a `decodeStrategy=None` for that use-case.

Reply:
I'm in agreement on` decodeStrategy`, and a `None` option makes sense as well, just to return the data as-is. 

For `metadataPolicy`, a bit confused about your pseudocode example. My understanding/thought process was this was an option to determine if the provider metadata should be combined into the Kubernetes metadata. So if AWS/Azure supports arbitrary tags on secrets, enabling this would combine those tags into the Kubernetes labels/annotations. Is my understanding of this wrong?

Reply:
I see. felt weird for me, too. That's what i understood in https://github.com/external-secrets/external-secrets/pull/22#discussion_r572075264. Feels like a misuse of tags TBH, but if there's a use-case then.. i don't know. I asked for feedback just to be 100% clear about that.

Reply:
Isn't it too early in the project life to start supporting this kind of super specific use case? Specially with this project not having any staging run time, and also not having sufficient unit/integration tests. I guess it does not hurt to foresee it on the spec, and leave the code ready for this to be implemented. But to have this already implemented together with the draft of a provider interface could be a problem, right? I am not sure tbh. Just thinking also of coming from an baby steps approach to make it easier to experiment/test and slowly grow features later. I may be missing something, I still need to properly read the other comments on the PR.